src/api/app/models/package.rb

```diff
@@ -117,13 +117,6 @@ class Package < ApplicationRecord
   has_one :backend_package, foreign_key: :package_id, dependent: :destroy, inverse_of: :package # rubocop:disable Rails/RedundantForeignKey
   has_many :tokens, dependent: :destroy
 
-  def self.check_access?(package)
-    return false if package.nil?
-    return false unless package.instance_of?(Package)
```

Comment:
I'm not sure if we can get rid of this condition. What happens if you try to do `package.project.check_access` on a `package` that returns something different from `Package`?

Package associated to a LinkedProject or similar:   `[https://github.com/openSUSE/open-build-service/pull/16768/commits/ae65d98712451d5849f89ddcad35d8d4c249bbb7#diff-9c1e6becaa519550173f515e411731cfd1d9810cdd4671c064627421445e0a76L122`](https://github.com/openSUSE/open-build-service/wiki/Links#instantiating-returns-nil-1)

Reply:
True, good catch. `package` could be a `String`. AFAIK, this only happens in one place, inside the `self.get_by_project_and_name` method.

I modified that method accordingly. Please take a look again.